server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,287 @@
+        for (int i = 0; i < stringKeyValue.length; i++) {
+            String[] keyValue = stringKeyValue[i].split(":");
+            settings.put(keyValue[0].trim(), keyValue[1].trim());
+        }
+
+        return settings.build();
+    }
+
+    private static RepositoryMetadata buildSegmentRepositoryMetadata(DiscoveryNode node) {
+        String name = node.getAttributes().get(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        String type = node.getAttributes().get(REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY);
+        String stringSettings = node.getAttributes().get(REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY);
+
+        assert name != null : "Name cannot be null";
+        assert type != null : "Type cannot be null";
+        assert stringSettings != null : "Settings cannot be null";
+
+        return new RepositoryMetadata(name, type, buildSettings(stringSettings));
+
+    }
+
+    private static RepositoryMetadata buildTranslogRepositoryMetadata(DiscoveryNode node) {
+        String name = node.getAttributes().get(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        String type = node.getAttributes().get(REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY);
+        String stringSettings = node.getAttributes().get(REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY);
+
+        assert name != null : "Name cannot be null";
+        assert type != null : "Type cannot be null";
+        assert stringSettings != null : "Settings cannot be null";
+
+        return new RepositoryMetadata(name, type, buildSettings(stringSettings));
+    }
+
+    /**
+     * Validated or adds the remote store repository to cluster state if it doesn't exist.
+     * @param joiningNode
+     * @param currentState
+     * @return updated cluster state
+     */
+    public static ClusterState validateOrAddRemoteStoreRepository(DiscoveryNode joiningNode, ClusterState currentState) {
+        List<DiscoveryNode> existingNodes = new ArrayList<>(currentState.getNodes().getNodes().values());
+
+        ClusterState newState = ClusterState.builder(currentState).build();
+
+        // TODO: Mutating cluster state like this can be dangerous, this will need refactoring.
+        if (existingNodes.size() == 0) {
+            // TODO: Will need to add the node to cluster state once the repository metadata is updated, such
+            // that subsequent joining nodes attributes gets validated against the existing nodes.
+            newState = updateClusterStateWithRepositoryMetadata(currentState, buildSegmentRepositoryMetadata(joiningNode));
+            newState = updateClusterStateWithRepositoryMetadata(newState, buildTranslogRepositoryMetadata(joiningNode));
+            return newState;
+        } else {
+            validateNodeAttributes(joiningNode, existingNodes.get(0));
+        }
+
+        return newState;
+    }
+
+    private static ClusterState updateClusterStateWithRepositoryMetadata(
```

Comment:
can we register multiple repository data in a single call? we would be adding another repo for remote cluster state as well, so we will be registering 3 repos in worst case.